### PR TITLE
Port over and update domain model from old blog post

### PR DESCRIPTION
**Pull request by marten**

### CODE REVIEW

File: domain_model.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Domain Model
+============
+
+The domain model is actually pretty simple. We typically think most about the following entities:
+
+User
+----
+
+People are core to the Zooniverse. When talking publically about the Zooniverse I almost always use the term ‘citizen scientist’ or ‘volunteer’ because it feels like an appropriate term for someone who donates their time to one of our projects. When writing code however, the shortest descriptive term that makes sense is usually selected so in our domain model the term we use is User.
```
Comment (alexbfree) [inline domain_model.rst]: Who is "I"? Should personalise the rest, or make it third person, I think.

File: domain_model.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+A User is exactly what you’d expect, it’s a person, it has a bunch of information associated with it such as a username, an email address, information about which projects they’ve helped with and a host of other bits and bobs. Crucially though for us, a User is the same regardless of which project they’re working – that is Users are pan-Zooniverse. Whether you’re classiying galaxies over at Galaxy Zoo or identifying animals on Snapshot Serengeti we’re associating your efforts with the same User record each time which turns out to be useful for a whole bunch of reasons (more later).
+
+Subject
+-------
+
+Just as people are core, as are the things that they’re analysing to help us do research. In Old Weather it’s a scanned image of a ship log book, in Planet Hunters it’s a light curve but regardless of the project internally we call all of these things Subjects. A Subject is the thing that we present to a User when we want to them to do something.
```
Comment (alexbfree) [inline domain_model.rst]: could do with a couple of extra commas in this second sentence. Bit long/unbroken.

File: domain_model.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Subjects are then stored in our database with a collection of metadata added from the manifest the user supplied while uploading. For example in Galaxy Zoo we might store some metadata associated with the survey telescope that imaged the galaxy and in Cyclone Center we store information about the date, time and position the image was recorded.
+
```
Comment (alexbfree) [inline domain_model.rst:19]: I would consider also taking a sentence to explain that a subject is not the same as the image... e.g. that some subjects have more than one image... or other things like video, etc.

File: domain_model.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Workflow
+--------
+
+The Workflow is another main entity in our system. It is where most of the parameters for a project are stored. Most importantly, the Workflow has the configuration of Tasks we ask our volunteers to perform when they are presented with a Subject. In Notes from Nature each step of the transcription (such as ‘What is the location?’) is a separate Task, in Galaxy Zoo, each step of the decision tree is a Task too. The Workflow also determines which Subjects are available to work on, and how long they should be worked on.
```
Comment (alexbfree) [inline domain_model.rst]: I think this needs a sentence upfront to actually say what a Workflow _is_. Something like: A workflow is a defined set of steps we ask every user to go through when they classify a subject.

Comment (alexbfree) [inline domain_model.rst]: Maybe give "Task" its own subheading? And talk  about questions vs drawing tasks?

File: domain_model.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+Classification
+--------------
+
+It’s no accident that I’ve introduced these three entities, User, Subject and Workflow first as the combination of these culminates in what we call a Classification. The Classification is the core unit of human effort produced by the Zooniverse community as it represents what a person saw and what they said about it. We collect a lot of these!
+
+SubjectSet
+----------
+
+Sometimes we need to group Subjects together for some higher level function. Perhaps it’s to represent a season’s worth of images in Snapshot Serengeti or a particular cell dye staining in Cell Slider. Whatever the reason for grouping, the entity we use to describe this is ‘SubjectSet’. The assignment of Subjects to Workflows actually happens by assigning SubjectSets to workflows, because a project often has hunderds of thousands of Subjects and assigning each of those individually would be tedious to do by hand.
+
```
Comment (alexbfree) [inline domain_model.rst:36]: Could add a section about Collections too, and differentiate from Subject Sets. Maybe Favorites too.

File: domain_model.rst [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+Workflow
+--------
+
+The Workflow is another main entity in our system. It is where most of the parameters for a project are stored. Most importantly, the Workflow has the configuration of Tasks we ask our volunteers to perform when they are presented with a Subject. In Notes from Nature each step of the transcription (such as ‘What is the location?’) is a separate Task, in Galaxy Zoo, each step of the decision tree is a Task too. The Workflow also determines which Subjects are available to work on, and how long they should be worked on.
+
+Classification
+--------------
+
+It’s no accident that I’ve introduced these three entities, User, Subject and Workflow first as the combination of these culminates in what we call a Classification. The Classification is the core unit of human effort produced by the Zooniverse community as it represents what a person saw and what they said about it. We collect a lot of these!
```
Comment (alexbfree) [inline domain_model.rst]: Another "I" is used here. Consider removing.

Comment (alexbfree) [inline domain_model.rst]: " the combination of these culminates in what we call a Classification" <== little bit abstract/vague. Maybe better to say.. As a User is presented with a Subject and progresses through the Workflow, they produce a Classification

Comment (alexbfree) [inline domain_model.rst]: Do we need to mention/explain the term "annotation" as well?

File: domain_model.rst [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+SubjectSet
+----------
+
+Sometimes we need to group Subjects together for some higher level function. Perhaps it’s to represent a season’s worth of images in Snapshot Serengeti or a particular cell dye staining in Cell Slider. Whatever the reason for grouping, the entity we use to describe this is ‘SubjectSet’. The assignment of Subjects to Workflows actually happens by assigning SubjectSets to workflows, because a project often has hunderds of thousands of Subjects and assigning each of those individually would be tedious to do by hand.
```
Comment (alexbfree) [inline domain_model.rst:35]: SubjectSet or Subject Set? Need to decide if we are talking about code (in which case use a class name) or concepts (in which case don't use camelcase)

File: domain_model.rst [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+Project
+-------
+
+The Zooniverse APIs host a whole bunch of different Zooniverse projects so it’s probably no surprise that we represent the actual citizen science project within our domain model. No prize for guessing the name of this entity – it’s called Project. A Project is really just the overarching named entity that SubjectSets and Workflows are associated with.
```
Comment (alexbfree) [inline domain_model.rst]: Opportunity to mention that this is what you create in the project builder? And that it has other associated things like FAQ/Research pages, description, avatar, Media, etc....?

**Comment by alexbfree** [comment]
Looks good. Have added some suggested improvements but none are critical.

File: domain_model.rst (hunk 1 above)
Comment (marten) [inline domain_model.rst]: Remnant of the old blog post I copied this from initially.

**Pull request merged**